### PR TITLE
Considering laser range_min when raytracing free space.

### PR DESCRIPTION
**Pull request by ZdenekM**
This is extremely useful when using simulated 2D laser from Velodyne (otherwise obstacles close to robot are raytraced and robot may hit something). I'm not sure if it's useful in general but might be...

I also added correct transformation into transformLaserScanToPointCloud -> if transformed into sensor frame then it doesn't make sense (one can use simpler projector). This should be correct and may improve costmap accuracy for robots moving at higher speeds :)

### CODE REVIEW

**Comment by DLu** [comment]
Two things.

1) Just for easier processing, can you please separate the transform stuff into a separate pull request? 

2) I understand your patch, but does the projectLaser call not filter out the min/max lasers already? https://github.com/ros-perception/laser_geometry/blob/hydro-devel/src/laser_geometry.cpp#L427

File: costmap_2d/plugins/obstacle_layer.cpp [hunk 1]
```diff
@@ -303,6 +311,12 @@ void ObstacleLayer::updateBounds(double origin_x, double origin_y, double origin
         continue;
       }
 
+      if (sq_dist >= laser_max_range_*laser_max_range_ /*|| sq_dist <= laser_min_range_*laser_min_range_*/)
```
Comment (DLu) [inline costmap_2d/plugins/obstacle_layer.cpp]: Please remove/fix the commented out code.

File: costmap_2d/plugins/obstacle_layer.cpp [hunk 2]
```diff
@@ -461,10 +481,23 @@ void ObstacleLayer::raytraceFreespace(const Observation& clearing_observation, d
...
+
+    double oox = laser_min_range_ * cos(al) + ox;
+    double ooy = laser_min_range_ * sin(al) + oy;
+
+    unsigned int x00 = x0;
+    unsigned int y00 = x0;
+
+    if (!worldToMap(oox, ooy, x00, y00))
+          continue;
+
```
Comment (DLu) [inline costmap_2d/plugins/obstacle_layer.cpp]: Due to the extra computation of this step, we may want to hide it behind a flag for people who don't care about range_min.

**Comment by ZdenekM** [comment]
Thanks for comments! I will try to separate it to two pull requests.

**Pull request closed without merging**